[Bookworm] Paramiko can't connect to DUT

<!--
     If you are reporting a new issue, make sure that we do not have any duplicates
     already open. You can ensure this by searching the issue list for this
     repository. If there is a duplicate, please close your issue and add a comment
     to the existing issue instead.

     If you suspect your issue is a bug, please edit your issue description to
     include the BUG REPORT INFORMATION shown below. If you fail to provide this
     information within 7 days, we cannot debug your issue and will close it. We
     will, however, reopen it if you later provide the information.

     For more information about reporting issues, see
     https://github.com/Azure/SONiC/wiki#report-issues

     ---------------------------------------------------
     GENERAL SUPPORT INFORMATION
     ---------------------------------------------------

     The GitHub issue tracker is for bug reports and feature requests.
     General support can be found at the following locations:

     - SONiC Support Forums - https://groups.google.com/forum/#!forum/sonicproject

     ---------------------------------------------------
     BUG REPORT INFORMATION
     ---------------------------------------------------
     Use the commands below to provide key information from your environment:
     You do NOT have to include this information if this is a FEATURE REQUEST
-->

#### Description

<!--
     Briefly describe the problem you are having in a few paragraphs.
-->

In the Bookworm upgrade, looks like Paramiko has some trouble connecting to the DUT, in `tacacs/test_accounting.py`

Test log: https://elastictest.org/scheduler/testplan/652625b1f773ca326e14c0fd?testcase=tacacs%2Ftest_accounting.py&type=console

#### Steps to reproduce the issue:
1.
2.
3.

#### Describe the results you received:


#### Describe the results you expected:

Snippet of log:

```
        if len(agreed_keys) == 0:
>           raise SSHException(
                "Incompatible ssh peer (no acceptable host key)"
            )  # noqa
E           paramiko.ssh_exception.SSHException: Incompatible ssh peer (no acceptable host key)

agreed_kex = ['curve25519-sha256@libssh.org', 'ecdh-sha2-nistp256', 'ecdh-sha2-nistp384', 'ecdh-sha2-nistp521', 'diffie-hellman-group16-sha512', 'diffie-hellman-group-exchange-sha256', ...]
agreed_keys = []
client_compress_algo_list = ['none', 'zlib@openssh.com']
client_encrypt_algo_list = ['chacha20-poly1305@openssh.com', 'aes128-ctr', 'aes192-ctr', 'aes256-ctr', 'aes128-gcm@openssh.com', 'aes256-gcm@openssh.com']
client_lang_list = ['']
client_mac_algo_list = ['umac-64-etm@openssh.com', 'umac-128-etm@openssh.com', 'hmac-sha2-256-etm@openssh.com', 'hmac-sha2-512-etm@openssh.com', 'hmac-sha1-etm@openssh.com', 'umac-64@openssh.com', ...]
```


#### Output of `show version`:

```
(paste your output here)
```
#### Output of `show techsupport`:

```
(paste your output here or download and attach the file here )
```

#### Additional information you deem important (e.g. issue happens only occasionally):

<!--
     Also attach debug file produced by `sudo generate_dump`
-->

--- a/dockers/docker-sonic-mgmt/Dockerfile.j2
+++ b/dockers/docker-sonic-mgmt/Dockerfile.j2
@@ -71,7 +71,7 @@ RUN pip3 install aiohttp \
                  netaddr \
                  netmiko==2.4.2 \
                  pandas \
-                 paramiko==2.7.1 \
+                 paramiko==2.9.5 \
                  passlib \
                  pexpect \
                  prettytable \

